module/data/item/templates/activities.mjs

```diff
@@ -184,27 +184,33 @@ export default class ActivitiesTemplate extends SystemDataModel {
     if ( foundry.utils.getType(source.uses?.recovery) !== "string" ) return;
 
     // If period is charges, set the recovery type to `formula`
-    if ( source.uses.per === "charges" ) {
+    if ( source.uses?.per === "charges" ) {
       if ( source.uses.recovery ) {
         source.uses.recovery = [{ period: "lr", type: "formula", formula: source.uses.recovery }];
-      } else {
+      } else if (source.uses) {
         delete source.uses.recovery;
       }
     }
 
     // If period is not blank, set an appropriate recovery type
-    else if ( source.uses.per ) {
-      if ( CONFIG.DND5E.limitedUsePeriods[source.uses.per]?.formula && source.uses.recovery ) {
-        source.uses.recovery = [{ period: source.uses.per, type: "formula", formula: source.uses.recovery }];
+    else if (source.uses?.per ) {
+      if ( CONFIG.DND5E.limitedUsePeriods[source.uses?.per]?.formula && source.uses?.recovery ) {
+        if(source.uses)
+        {
+          source.uses.recovery = [{ period: source?.uses?.per, type: "formula", formula: source?.uses?.recovery }];
+        }
+      }
+      else if(source.uses)
+      {
+        source.uses.recovery = [{ period: source?.uses?.per, type: "recoverAll" }];
       }
```

Comment:
```suggestion
    else if ( source.uses.per ) {
      if ( CONFIG.DND5E.limitedUsePeriods[source.uses.per]?.formula && source.uses.recovery ) {
        source.uses.recovery = [{ period: source.uses.per, type: "formula", formula: source.uses.recovery }];
      }
      else source.uses.recovery = [{ period: source.uses.per, type: "recoverAll" }];
```
`source.uses` is (at this point) guaranteed to exist, so can remove all optional chaining/checking for it.